packages/devextreme/testing/tests/DevExpress.ui.widgets/chat.tests.js

```diff
@@ -371,6 +370,65 @@ QUnit.module('renderMessage', moduleConfig, () => {
 
         assert.strictEqual($(lastBubble).text(), text);
     });
+
+    QUnit.test('Message list should not run invalidate if 1 new message has been added to items', function(assert) {
+        const invalidateStub = sinon.stub(this.instance._messageList, '_invalidate');
+
+        const { items } = this.instance.option();
+        const newMessage = {
+            timestamp: NOW,
+            author: userFirst,
+            text: 'NEW MESSAGE',
+        };
+
+        this.instance.option({ items: [...items, newMessage] });
+
+        assert.strictEqual(invalidateStub.callCount, 0);
+    });
+
+    QUnit.test('Message list should run renderMessage if 1 new message has been added to items', function(assert) {
+        const renderMessageStub = sinon.stub(this.instance._messageList, '_renderMessage');
+
+        const { items } = this.instance.option();
+        const newMessage = {
+            timestamp: NOW,
+            author: userFirst,
+            text: 'NEW MESSAGE',
+        };
+
+        this.instance.option({ items: [...items, newMessage] });
+
+        assert.strictEqual(renderMessageStub.callCount, 1);
+    });
+
+    QUnit.test('Message list should run invalidate if all items was updated, but the items length the same', function(assert) {
```

Comment:
According to your code it should be
```suggestion
    QUnit.test('Message list should run invalidate if new items length is the same as current items length', function(assert) {
```

Reply:
[dbaeffb0202135fa22aac1fc8a6b2c01df2b7804]